Sources/Sentry/SentryBreadcrumbTracker.m

```diff
@@ -23,49 +25,62 @@ - (void)start
 
 - (void)trackApplicationUIKitNotifications
 {
+#if SENTRY_HAS_UIKIT
+    NSNotificationName foregroundNotificationName = UIApplicationDidBecomeActiveNotification;
+    NSNotificationName backgroundNotificationName = UIApplicationDidEnterBackgroundNotification;
+#elif TARGET_OS_OSX || TARGET_OS_MACCATALYST
+    NSNotificationName foregroundNotificationName = NSApplicationDidBecomeActiveNotification;
+    // Will resign Active notification is the nearest one to UIApplicationDidEnterBackgroundNotification
+    NSNotificationName backgroundNotificationName = NSApplicationWillResignActiveNotification;
+#else
+    [SentryLog logWithMessage:@"NO UIKit -> [SentryBreadcrumbTracker "
+     @"trackApplicationUIKitNotifications] does nothing."
+                     andLevel:kSentryLogLevelDebug];
+#endif
+    
+// not available for macOS
 #if SENTRY_HAS_UIKIT
     [NSNotificationCenter.defaultCenter
-        addObserverForName:UIApplicationDidReceiveMemoryWarningNotification
-                    object:nil
-                     queue:nil
-                usingBlock:^(NSNotification *notification) {
-                    if (nil != [SentrySDK.currentHub getClient]) {
-                        SentryBreadcrumb *crumb =
-                            [[SentryBreadcrumb alloc] initWithLevel:kSentryLevelWarning
-                                                           category:@"device.event"];
-                        crumb.type = @"system";
-                        crumb.data = @ { @"action" : @"LOW_MEMORY" };
-                        crumb.message = @"Low memory";
-                        [SentrySDK addBreadcrumb:crumb];
-                    }
-                }];
-
+     addObserverForName:UIApplicationDidReceiveMemoryWarningNotification
```

Comment:
`h`: I think we should add tests for this class here.

Reply:
@philipphofmann which class do you mean? `addObserverForName:UIApplicationDidReceiveMemoryWarningNotification` is a UIKit class

Reply:
Sorry, I could have added this information: `SentryBreadcrumbTracker`

Reply:
this class has been formatted by the IDE, it didn't change its logic, but yes, I've added just a few flags to make it to compile on all the platforms.
This should be tested of course, but not on this PR though, all the tests of this class have been removed on the v5 migration and all of them should be fixed altogether.